tools/serverpod_cli/lib/src/generator/dart/library_generators/doc_comments/build_doc_comment.dart

```diff
@@ -0,0 +1,32 @@
+extension PrependSlashes on String {
+  String prependSlashes() {
+    return split('\n').map((String line) => '/// $line').join('\n');
+  }
+}
+
+String buildDocComment(
+  String generalDescription,
+  Map<String, String> docByParameter,
+  List<String> parameters,
+) {
+  var index = 0;
+  var numberOfParameters = parameters.length;
+  var parameterDocComments =
+      parameters.fold('', (String previousValue, parameter) {
+    var parameterDoc = docByParameter[parameter];
+    if (parameterDoc == null) {
+      throw StateError('No documentation for parameter $parameter');
+    }
```

Comment:
Question: What is the reason for using two parameters here, `docByParameter` and `parameters`?  Couldn't we simply have a single Map that contains the parameters? Then the user of the interface is responsible for baking the list and we don't have a direct dependency towards all parameters.

Reply:
[42295ce](https://github.com/serverpod/serverpod/pull/2921/commits/42295cecdd4bd7889d9612ec6fde04c12fb6c248)